src/Users.js

```diff
@@ -212,29 +212,38 @@ class Users extends React.Component {
   }
 
   // XXX something prevents exceptions in this function from being received: see STRIPES-483
-  create = (userdata) => {
+  create = async (userdata) => {
     const { mutator } = this.props;
-    if (userdata.username) {
-      const creds = Object.assign({}, userdata.creds, { username: userdata.username }, userdata.creds.password ? {} : { password: '' });
-      const user = Object.assign({}, userdata, { id: uuid() });
-      if (user.creds) delete user.creds;
-
-      mutator.records.POST(user)
-        .then(newUser => mutator.creds.POST(Object.assign(creds, { userId: newUser.id })))
-        .then(newCreds => mutator.perms.POST({ userId: newCreds.userId, permissions: [] }))
-        .then((perms) => {
-          mutator.query.update({ _path: `/users/view/${perms.userId}`, layer: null });
-        });
-    } else {
-      const user = Object.assign({}, userdata, { id: uuid() });
-      if (user.creds) delete user.creds;
-
-      mutator.records.POST(user)
-        .then((newUser) => mutator.perms.POST({ userId: newUser.id, permissions: [] }))
-        .then((perms) => {
-          mutator.query.update({ _path: `/users/view/${perms.userId}`, layer: null });
-        });
-    }
+    let { username } = userdata;
+    username = username.trim();
+    const createdUserdata = {
+      ...userdata,
+      username
+    };
+    const creds = {
+      ...createdUserdata.creds,
+      username: createdUserdata.username,
+      password: userdata.creds.password
+    };
+    const user = {
+      ...createdUserdata,
+      id: uuid()
+    };
+    if (user.creds) delete user.creds;
+    const newUser = await mutator.records.POST(user);
```

Comment:
You should get `POST` from destructuring props. Also, it might be a good idea to rename it during destructuring.